numpy/core/src/multiarray/array_method.h

```diff
@@ -203,6 +226,8 @@ typedef struct PyArrayMethodObject_tag {
     PyArray_DTypeMeta **wrapped_dtypes;
     translate_given_descrs_func *translate_given_descrs;
     translate_loop_descrs_func *translate_loop_descrs;
+    /* Chunk used by the legacy fallback arraymethod mainly */
+    char initial[sizeof(npy_clongdouble)];  /* initial value storage */
```

Comment:
True that it is not quite general - still, since the allocation is done elsewhere and data is copied from this location, I'm not sure there is a net saving in complexity in having what seems a rather odd difficult to explain piece of `PyArrayMethodObject_tag` -- it just doesn't feel properly extensible/re-usable, unlike the rest.

Reply:
This is the one part that still feels super clunky. I complain in part because the rest feels really nice and logical yet here were suddenly have some opaque blob of bytes! 

Looking at the code a bit more, a pointer to some memory allocated on first call would not be difficult right? You'd just have to be careful to deallocate it in `arraymethod_dealloc`, like now done with `name`.

Reply:
No, it would be easy to do such a pointer.  Maybe our difference is that I still consider `PyArrayMethodObject` to be fully opaque and private to NumPy?

Reply:
I guess the basic issue remains that having a chunk of random bytes feels clunky, but you are right that to me the struct really seems something that could become public (or, if not, a really useful thing for developers to look at), so one should keep it neat. And if it becomes public, having a pointer seems far more useful. Probably even better if it is a pointer to something well-defined like an array scalar.

Reply:
Hmmm, it is filled always with the same value and when the GIL is held, so I think it is fine.  Will adept the comment to "only" not "mainly".  Because you are right: reusing it outside of the "legacy" path might be prone to threadsafety issues.